### PR TITLE
feat(flags): add MemberFlags and relevant fields

### PR DESCRIPTION
**Pull request by Soheab**
## Summary

Relevant link to API Docs:
- https://discord.com/developers/docs/resources/guild#guild-member-object-guild-member-structure
- https://discord.com/developers/docs/resources/guild#guild-member-object-guild-member-flags

This PR adds support for getting a member's flags via the matching attributes.

I have not tested this but.. most of the code is from **my** PR to discord.py - https://github.com/Rapptz/discord.py/pull/9204 which was fully tested.

Adds the following:

- Classes:
  - `MemberFlags`

- Attributes to nextcord.Member:
  - `_flags`
  - `flags` (property)

- Kwargs to nextcord.Member.edit:
  - `flags`
  - `bypass_verification`

- Fields to `PartialMember` (type):
  - `flags`

- Possible attrs for `on_member_update` (docs):
  - `flags`

## Checklist

- [ ] If code changes were made then they have been tested.
  - [x] I have updated the documentation to reflect the changes.
  - [x] I have run `task pyright` and fixed the relevant issues.
- [ ] This PR fixes an issue.
- [x] This PR adds something new (e.g. new method or parameters).
- [ ] This PR is a breaking change (e.g. methods or parameters removed/renamed)
- [ ] This PR is **not** a code change (e.g. documentation, README, ...)

### CODE REVIEW

File: nextcord/flags.py [hunk 1]
```diff
@@ -486,6 +487,79 @@ def all(self) -> List[UserFlags]:
...
+
+        .. describe:: x == y
+
+            Checks if two MemberFlags are equal.
+
+        .. describe:: x != y
+
+            Checks if two MemberFlags are not equal.
+
+        .. describe:: x | y, x |= y
```
Comment (ooliver1) [inline nextcord/flags.py:506]: `__xor__`, `__invert__`, `__and__`, `__or__` and the relevant in-place variants currently are not implemented for flag values.

**Review by ooliver1 — APPROVED** [review]

**Review by ooliver1 — CHANGES_REQUESTED** [review]
Accidental approval yet again

File: nextcord/flags.py [hunk 2]
```diff
@@ -486,6 +487,58 @@ def all(self) -> List[UserFlags]:
...
+    Attributes
+    -----------
+    value: :class:`int`
+        The raw value. You should query flags via the properties
+        rather than using this raw value.
+    """
+
+    @flag_value
+    def did_rejoin(self):
+        """:class:`bool`: Returns ``True`` if the member left and rejoined the :attr:`~nextcord.Member.guild`."""
```
Comment (EmreTech) [inline nextcord/flags.py:523]: Why are you referencing the member's guild attribute and not the guild class?

File: nextcord/member.py [hunk 3]
```diff
@@ -693,28 +707,32 @@ async def edit(
...
+        +---------------------+--------------------------------------+
+        | deafen              | :attr:`Permissions.deafen_members`   |
+        +---------------------+--------------------------------------+
+        | roles               | :attr:`Permissions.manage_roles`     |
+        +---------------------+--------------------------------------+
+        | voice_channel       | :attr:`Permissions.move_members`     |
+        +---------------------+--------------------------------------+
+        | timeout             | :attr:`Permissions.moderate_members` |
+        +---------------------+--------------------------------------+
+        | bypass_verification | :attr:`Permissions.moderate_members` |
```
Comment (EmreTech) [inline nextcord/member.py:734]: What about the flags parameter?

File: nextcord/flags.py [hunk 4]
```diff
@@ -486,6 +487,58 @@ def all(self) -> List[UserFlags]:
         return [public_flag for public_flag in UserFlags if self._has_flag(public_flag.value)]
 
 
+@fill_with_flags()
+class MemberFlags(BaseFlags):
+    r"""Wraps up the Discord Guild Member flags
+
+    .. versionadded:: 2.5
```
Comment (EmreTech) [inline nextcord/flags.py:494]:
```suggestion
    .. versionadded:: 2.6
```

File: nextcord/member.py [hunk 5]
```diff
@@ -544,6 +550,14 @@ def activity(self) -> Optional[ActivityTypes]:
         if self.activities:
             return self.activities[0]
 
+    @property
+    def flags(self) -> MemberFlags:
+        """:class:`MemberFlags`: Returns the member's flags.
+
+        .. versionadded:: 2.5
```
Comment (EmreTech) [inline nextcord/member.py:557]:
```suggestion
        .. versionadded:: 2.6
```

File: nextcord/member.py [hunk 6]
```diff
@@ -749,6 +767,15 @@ async def edit(
             Set this to None to disable their timeout.
 
             .. versionadded:: 2.0
+        flags: :class:`~nextcord.MemberFlags`
+            The flags to set for this member.
+            Currently only :class:`~nextcord.MemberFlags.bypasses_verification` is able to be set.
+
+            .. versionadded:: 2.5
```
Comment (EmreTech) [inline nextcord/member.py:774]:
```suggestion
            .. versionadded:: 2.6
```

File: nextcord/member.py [hunk 7]
```diff
@@ -749,6 +767,15 @@ async def edit(
...
+        bypass_verification: :class:`bool`
+            Indicates if the member should be allowed to bypass the guild verification requirements.
+
+            .. versionadded:: 2.5
```
Comment (EmreTech) [inline nextcord/member.py:778]:
```suggestion
            .. versionadded:: 2.6
```

**Review by EmreTech — CHANGES_REQUESTED** [review]

File: nextcord/member.py [hunk 8]
```diff
@@ -282,6 +284,7 @@ def __init__(
         self._timeout: Optional[datetime.datetime] = utils.parse_time(
             data.get("communication_disabled_until")
         )
+        self._flags = data.get("flags", 0)
```
Comment (ooliver1) [inline nextcord/member.py:287]:
```suggestion
        self._flags: int = data.get("flags", 0)
```
like all other attributes

File: nextcord/member.py (hunk 3 above)
Comment (ooliver1) [reply nextcord/member.py:734]: Well that depends on the flags set.

**Review by ooliver1 — CHANGES_REQUESTED** [review]

File: nextcord/member.py (hunk 3 above)
Comment (EmreTech) [reply nextcord/member.py:734]: Good point.

File: nextcord/flags.py (hunk 2 above)
Comment (Soheab) [reply nextcord/flags.py:523]: Because that's the relevant guild I think.

Comment (EmreTech) [reply nextcord/flags.py:523]: It doesn't make sense to reference the member's guild attribute here. Just refer to the guild class instead.

Comment (Soheab) [reply nextcord/flags.py]: Done!

File: nextcord/flags.py [hunk 9]
```diff
@@ -486,6 +487,58 @@ def all(self) -> List[UserFlags]:
...
+        .. describe:: x == y
+
+            Checks if two MemberFlags are equal.
+
+        .. describe:: x != y
+
+            Checks if two MemberFlags are not equal.
+        .. describe:: hash(x)
+
+            Return the flag's hash.
```
Comment (ooliver1) [inline nextcord/flags.py:507]:
```suggestion
            Returns the flag's hash.
```
Same form as the other operations: "Returns", "Checks".

**Review by ooliver1 — APPROVED** [review]

**Review by EmreTech — APPROVED** [review]

**Pull request merged**